dataset/src/test/scala/frameless/functions/UdfTest.scala

```diff
@@ -0,0 +1,194 @@
+package frameless
+package functions
+
+import org.scalacheck.Prop
+import org.scalacheck.Prop._
+
+class UdfTest extends TypedDatasetSuite {
+
+  test("one argument udf") {
+    def prop[A, B, C](data: Vector[X3[A, B, C]], f1: A => A)(
+      implicit
+      ea: TypedEncoder[A],
+      eb: TypedEncoder[B],
+      ec: TypedEncoder[C],
+      ex4: TypedEncoder[X3[A, B, C]]
+    ): Prop = {
+      val dataset = TypedDataset.create(data)
+      val u1 = udf[X3[A, B, C], A, A](f1)
+      val u2 = dataset.makeUDF(f1)
```

Comment:
It there any difference between `udf` and `makeUDF`? I feel it would be enough to test only `makeUDF`, it would make test code more readable.